pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -101,18 +114,24 @@ func (e *Executor) checkApproval(ctx context.Context) (string, bool) {
 		return "", false
 	}
 
-	metadata := map[string]string{
-		approvedByKey: approveCmd.Commander,
+	if err := approveCmd.Report(ctx, model.CommandStatus_COMMAND_SUCCEEDED, nil, nil); err != nil {
```

Comment:
Now I'm wondering if it's fine to report as SUCCEEDED even if `e.MetadataStore.Stage(e.Stage.Id).PutMulti(ctx, metadata)` got failed.

Reply:
Yeah, I'm concerned about it, too.

Reply:
But I can't figure out it so far.